### PR TITLE
Semantic metadata

### PR DESCRIPTION
**Pull request by dotmancando**
## Update TestHarness to thingfulx version semantic-metadata
Changes include:
* Generally change "fetcher" to indexer"
* Remove `Provider()`, use `UID()` instead
* Clear separation of `Fetch()` and `Parse()`, although we don't separately run them in `testharness` yet

### CODE REVIEW

File: testharness.go [hunk 1]
```diff
@@ -143,7 +152,14 @@ func (h *Harness) RunAll(ctx context.Context, fetchInterval time.Duration, total
...
+		bytes, err := h.indexer.Fetch(ctx, u, clientFetch)
+		if err != nil {
+			fmt.Printf("## ERROR from Fetch: %s\n", err.Error())
+			FetchError = append(FetchError, err.Error())
+			FetchErrorCount += 1
+		}
+
+		things, err := h.indexer.Parse(bytes, u, timeProvider)
 		if err != nil {
 			fmt.Printf("## ERROR from Fetch: %s\n", err.Error())
```
Comment (Popcore) [inline testharness.go]: This should say "ERROR from Parse" now

File: testharness.go [hunk 2]
```diff
@@ -228,7 +244,16 @@ func (h *Harness) RunFetch(ctx context.Context, urls []string, fetchInterval tim
...
+		if err != nil {
+			fmt.Printf("## ERROR from Fetch: %s\n", err.Error())
+			FetchError = append(FetchError, err.Error())
+			FetchErrorCount += 1
+		}
+
+		things, err := h.indexer.Parse(bytes, u, timeProvider)
+
 		if err != nil {
 			fmt.Printf("## ERROR from Fetch: %s\n", err.Error())
```
Comment (Popcore) [inline testharness.go]: This should say "ERROR from Parse" too

File: testharness.go [hunk 3]
```diff
@@ -291,7 +316,16 @@ func (h *Harness) RunAccess(ctx context.Context, urls []string, fetchInterval ti
...
```
Comment (Popcore) [inline testharness.go]: This should say "ERROR from Parse" too

**Comment by smulube** [comment]
maybe consider running gometalinter on the code - bunch of little tweaks it will recommend i think, no errors but almost certainly a few changes to make this more idiomatic (god i'm starting to hate that word ;)

**Review by smulube — CHANGES_REQUESTED** [review]
see my comment @dotmancando, oh and @Popcore's

**Review by smulube — APPROVED** [review]

File: testharness.go [hunk 4]
```diff
@@ -57,7 +77,7 @@ func (h *Harness) RunAll(ctx context.Context, fetchInterval time.Duration, total
 		// panic(err)
```
Comment (dotmancando) [inline testharness.go]: remove

File: glide.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@ package: github.com/thingful/testharness
 import:
 - package: github.com/thingful/httpmock
 - package: github.com/thingful/thingfulx
-  version: 0.1.9
+  version: 0.2.0
```
Comment (smulube) [inline glide.yaml]: is this the correct version?

Comment (smulube) [reply glide.yaml]: glide does have concept of patch ranges, so you could change this to:

```
version: ~0.2.0
```

which is the same as: `>= 0.2.0 && < 0.3.0`

**Pull request merged**